### PR TITLE
DBTP-659 - Add deletion-protection property to copilot-tools

### PR DESCRIPTION
**Pull request by JohnStainsby**
- Add deletion-protection property for aurora and rds postgres addons

### CODE REVIEW

File: addons-example.yml [hunk 1]
```diff
@@ -90,8 +92,11 @@ my-rds-db:
   environments:
     default:
       plan: small-ha
-      deletion-policy: Retain # Delete|Retain|Snapshot
```
Comment (yusufsheiqh) [inline addons-example.yml:93]: Do we still need this policy? Or does protection take priority?

Comment (yusufsheiqh) [reply addons-example.yml:93]: I can see both are set for the `production` env.

Comment (JohnStainsby) [reply addons-example.yml:93]: The `deletion-policy` is specific to preventing CloudFormation deleting a resource. Presumably `deletion-protection` would also stop that scenario but I'm not sure. There may be situations where you'd want one vs the other.

File: addons-example.yml [hunk 2]
```diff
@@ -11,11 +11,13 @@
 #       some-setting: x
 #       another-setting: y
 #       plan: a-valid-plan  # some addon types support plans/t-shirt sizes.  These are listed in addon-plans.yaml in this repo.
+#       deletion-protection: X # true|false rds-postgres and aurora-postgres only
```
Comment (WillGibson) [inline addons-example.yml:14]: If this is an example, perhaps...
```suggestion
#       deletion-protection: false # true|false rds-postgres and aurora-postgres only
```

File: addons-example.yml (hunk 1 above)
Comment (WillGibson) [reply addons-example.yml:93]: It does feel a bit weird having different settings overlapping like this, but I think we can thank AWS for that.

Comment (JohnStainsby) [reply addons-example.yml:93]: It is strange but I don't think there's much we can do about it. I tested it and `deletion-protection` does stop CloudFormation from deleting the database but it gets stuck in _update-cleanup-in-progress_ for ages.

Also `deletion-protection` is only an option for databases so `deletion-policy` would still have to be set on the other addons anyway.

**Review by WillGibson — APPROVED** [review]

**Pull request merged**